src/compat.h

```diff
@@ -0,0 +1,38 @@
+#ifndef TIMESCALEDB_COMPAT_H
```

Comment:
There are inevitably places in the code that require `#if PG_VERSION >= ...`. I put macro wrappers here for functions that have a simple signature change and which would otherwise became particularly messy. I tried to avoid moving big chunks of code into, e.g.,  `compat.c`, because I think you lose context by doing that. I think there's value to seeing the "diff" between versions/APIs where the code is used. For instance, I see no reason to move the incompatible `hooks` (like the process utility hook) into `compat.c/h` as a wrapper since it only requires a change in one place (i.e., we do not hook into it in more than one place) and everything that deals with process utility stuff is in `process_utility.c`.

I think it is hard to come up with a systematic rule for this stuff. I think we have to be guided by common sense and strive towards making the code clean and understandable.

Reply:
ok